crates/iceberg-catalog/src/catalog/config.rs

```diff
@@ -60,16 +60,25 @@ impl<A: Authorizer + Clone, C: Catalog, S: SecretStore>
             .await?;
 
         // Get config from DB and new token from AuthHandler simultaneously
-        let mut config =
-            C::require_config_for_warehouse(warehouse_id, api_context.v1_state.catalog).await?;
+        let mut config = C::require_config_for_warehouse(
+            warehouse_id,
+            request_metadata.host(),
+            api_context.v1_state.catalog,
+        )
+        .await?;
 
         config
             .overrides
             .insert("prefix".to_string(), CONFIG.warehouse_prefix(warehouse_id));
 
-        config
-            .overrides
-            .insert("uri".to_string(), CONFIG.base_uri_catalog().to_string());
+        config.overrides.insert(
+            "uri".to_string(),
+            request_metadata
```

Comment:
We should switch the priority: Explicit config comes before Header parsing for a few reasons:
1. Backwards compatibility: This change could break pretty much all current deployments that are behind a proxy that is not configured to set the headers. `BASE_URL` is set correctly for those deployments. Who knows if headers are set correctly.
2. Deployments don't always have the proxy configuration in their hands, but they can configure Lakekeeper freely. Thus users need a way to force something on Lakekeeper side - our `BASE_URL` env var.

When changing this, we also need to change the docs.

Reply:
Then we'll either have to remove the default of base_url which is a breaking change, or we'll have to add a flag which disables the base_location default which also doesn't sound too pretty